File: src/main/java/kr/mjc/rachel/spring/jdbc/user/GetUser.java
package kr.mjc.rachel.spring.jdbc.user;

import kr.mjc.rachel.basics.jdbc.user.User;
import kr.mjc.rachel.basics.jdbc.user.dao.UserDao;
import kr.mjc.rachel.spring.jdbc.AppTest;
import org.slf4j.Logger;
import org.springframework.dao.DataAccessException;

import java.util.Scanner;

/**
 * userId로 회원정보를 가져온다.
 */
public class GetUser {
  public static void main(String[] args) {
    new AppTest().test(UserDao.class, (UserDao userDao, Logger log) -> {
      Scanner scanner = new Scanner(System.in);
      System.out.print("Get - userId : ");
      int userId = scanner.nextInt();
      scanner.close();
      try {
        User user = userDao.getUser(userId);
        log.info(user.toString());
      } catch (DataAccessException e) { // 데이터 없을 경우
        log.error("회원 없음");
      }
    });
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/jdbc/user/GetUser.kt
package kr.mjc.jacob.spring.jdbc.user

import kr.mjc.jacob.basics.jdbc.user.dao.UserDao
import kr.mjc.jacob.spring.jdbc.AppTest
import org.springframework.dao.DataAccessException
import java.util.*

fun main() {
  AppTest().test(UserDao::class.java) { userDao, log ->
    print("Get - userId ? ")
    val userId = Scanner(System.`in`).use { it.nextInt() }
    try {
      val user = userDao.getUser(userId)
      log.debug(user.toString())
    } catch (e: DataAccessException) {
      log.error("회원 없음: userId=$userId")
    }
  }
}
